app/assets/javascripts/member-facing/redirectors.js

```diff
@@ -0,0 +1,43 @@
+const RegisterMemberRedirector = {
+  attemptRedirect:  function(followUpUrl, member) {
+
+    if(typeof(member) !== 'object') {
+      member = window.champaign.personalization.member;
+    }
+
+    if ( shouldMemberRegister() ) {
+      redirectToRegistration();
+      return true;
+    } else {
+      return false;
+    }
+
+    function shouldMemberRegister() {
```

Comment:
Thanks @eyko. I don't really like the idea of writing the functions outside of `RegisterMemberRedirector` since they are helper functions only used by this redirector. 

I also see you're using urijs, where do we stand on this? Do you want me to remove it or not?